File: src/services/vault.service.ts
import { supabase } from '../lib/supabase/client';

export async function listVaultEntries(userId: string) {
  const { data, error } = await supabase
    .from('vault_entries')
    .select('*')
    .eq('user_id', userId)
    .order('created_at', { ascending: false });
  if (error) throw error;
  return data ?? [];
}

export async function registerVaultIP(params: {
  user_id: string;
  node_id?: string;
  vault_entry_id?: string;
  version_hash: string;
  ownership_signature: string;
}) {
  const { data, error } = await supabase.from('ai_innovation_vault').insert(params).select().single();
  if (error) throw error;
  return data;
}

File: mobile_app/lib/features/vault/services/vault_service.dart
import 'package:supabase_flutter/supabase_flutter.dart';
import '../../../core/supabase_client.dart';
import '../models/vault_entry.dart';

class VaultService {
  final SupabaseClient _client = SupabaseConfig.client;

  Future<List<VaultEntry>> listVaultEntries({String? userId}) async {
    var query = _client.from('vault_entries').select('*');

    if (userId != null) query = query.eq('user_id', userId);

    final data = await query.order('created_at', ascending: false);
    return (data as List<dynamic>)
        .map((json) => VaultEntry.fromJson(json as Map<String, dynamic>))
        .toList();
  }

  Future<VaultEntry> getVaultEntry(String id) async {
    final data = await _client.from('vault_entries').select('*').eq('id', id).single();
    return VaultEntry.fromJson(data);
  }

  Future<VaultEntry> createVaultEntry(VaultEntry entry) async {
    final data = await _client
        .from('vault_entries')
        .insert(entry.toJson())
        .select()
        .single();
    return VaultEntry.fromJson(data);
  }

  Future<void> updateVaultEntry(String id, Map<String, dynamic> updates) async {
    await _client.from('vault_entries').update(updates).eq('id', id);
  }

  Future<void> deleteVaultEntry(String id) async {
    await _client.from('vault_entries').delete().eq('id', id);
  }
}
